### PR TITLE
Clean up docker container rule

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[ERROR] src/main/java/org/apache/james/utils/FakeSmtp.java:[33,8] (imports) UnusedImports: Unused import - org.awaitility.core.ConditionFactory
```

File: server/testing/src/main/java/org/apache/james/util/docker/DockerGenericContainer.java [hunk 1]
```diff
@@ -39,39 +39,35 @@
...
+        return new DockerGenericContainer(new GenericContainer<>(imageFromDockerfile));
+    }
+
+    private static void disableDockerTestsIfDockerUnavailable() {
         try {
-            this.container = new GenericContainer<>(imageFromDockerfile);
+            DockerClientFactory.instance().client();
         } catch (IllegalStateException e) {
-            logAndCheckSkipTest(e);
+            LOGGER.error("Cannot initial a docker container", e);
```
Comment (Arsnael) [inline server/testing/src/main/java/org/apache/james/util/docker/DockerGenericContainer.java]: s/initial/initialize

**Comment by trantienduchn** [comment]
same above in ES mpt module
```
[b01c73a3c5fb405e4c454485ee73afe095e8c94c] [ERROR] Failures: 
[b01c73a3c5fb405e4c454485ee73afe095e8c94c] [ERROR]   UidSearchOnIndex.testSearchAtomsUS Multiple Failures (2 failures)
[b01c73a3c5fb405e4c454485ee73afe095e8c94c] 	org/junit/AssumptionViolatedException
[b01c73a3c5fb405e4c454485ee73afe095e8c94c] 	<no message> in java.lang.NullPointerException
```

**Comment by chibenwa** [comment]
test this please

**Comment by mbaechler** [comment]
test this please

**Comment by mbaechler** [comment]
test this please

**Comment by Arsnael** [comment]
Lost the output during the weekend....

test this please

**Comment by mbaechler** [comment]
```
[4af9daf73b2e7b25ea9c0f1b7ca1fa27da405af6] [ERROR] Errors: 
[4af9daf73b2e7b25ea9c0f1b7ca1fa27da405af6] [ERROR]   SetMessagesMethodTest.setup:162 » ElasticsearchStatus Elasticsearch exception ...
```

test this please

**Comment by mbaechler** [comment]
```
[4af9daf73b2e7b25ea9c0f1b7ca1fa27da405af6] [ERROR] Errors: 
[4af9daf73b2e7b25ea9c0f1b7ca1fa27da405af6] [ERROR]   RabbitMQEventBusTest$LifeCycleTest$SingleEventBus$DispatchingWhenNetWorkIssue.dispatchShouldWorkAfterNetworkIssuesForOldRegistration:366 » ConditionTimeout
```
test this please

**Comment by mbaechler** [comment]
```
[4af9daf73b2e7b25ea9c0f1b7ca1fa27da405af6] [ERROR] Errors: 
[4af9daf73b2e7b25ea9c0f1b7ca1fa27da405af6] [ERROR]   RabbitMQEventBusTest$LifeCycleTest$SingleEventBus$DispatchingWhenNetWorkIssue.dispatchShouldWorkAfterNetworkIssuesForOldRegistration:366 » ConditionTimeout
```
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by blackheaven** [comment]
test this please

File: server/testing/src/main/java/org/apache/james/utils/FakeSmtp.java [hunk 2]
```diff
@@ -56,30 +56,31 @@ public static void clean(RequestSpecification requestSpecification) {
...
+    public static FakeSmtp withDefaultPort() {
+        return withSmtpPort(SMTP_PORT);
+    }
+
+    private static DockerContainer fakeSmtpContainer() {
+        return DockerContainer.fromName(Images.FAKE_SMTP)
             .withAffinityToContainer()
             .waitingFor(new HostPortWaitStrategy()
-            .withRateLimiter(RateLimiters.TWENTIES_PER_SECOND));
+                .withRateLimiter(RateLimiters.TWENTIES_PER_SECOND)
```
Comment (chibenwa) [inline server/testing/src/main/java/org/apache/james/utils/FakeSmtp.java]: Do we have an indent issue?

Comment (blackheaven) [reply server/testing/src/main/java/org/apache/james/utils/FakeSmtp.java]: we do not, it's in `waitingFor`

**Pull request merged**